core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -3628,8 +3628,8 @@ public int compare( Object data1, Object data2 ) throws KettleValueException {
         // if (isStorageBinaryString() && identicalFormat &&
         // storageMetadata.isSingleByteEncoding()) return
         // compareBinaryStrings((byte[])data1, (byte[])data2); TODO
-        String one = getString( data1 );
-        String two = getString( data2 );
+        String one = getString( data1 ).trim();
+        String two = getString( data2 ).trim();
 
```

Comment:
This'll change more than just sorting of step metrics.  Do we know for sure that leading and tailing white space should be ignored in all cases?